### PR TITLE
T4242585 Enable tiered storage only for non compacted topics.

### PR DESCRIPTION
**Pull request by satishd**
T4242585 Enable tiered storage only for non compacted topics.

### CODE REVIEW

**Comment by satishd** [comment]
^^ @ying-zheng

File: core/src/main/scala/kafka/log/Log.scala [hunk 1]
```diff
@@ -277,6 +277,13 @@ class Log(@volatile var dir: File,
 
   @volatile private var localLogStartOffset:Long = logStartOffset
 
+  @volatile private var highestOffsetWithRemoteIndex:Long = logStartOffset
+
+  private def effectiveRemoteLogEnabled() : Boolean = {
```
Comment (harshach) [inline core/src/main/scala/kafka/log/Log.scala]: lets just all call it remoeLogEnabled and add few comments saying we will ignore the compacted topics

Comment (satishd) [reply core/src/main/scala/kafka/log/Log.scala]: okay, addressed with the latest commit.

**Comment by harshach** [comment]
+1

**Pull request closed without merging**